src/app.js

```diff
@@ -6,8 +6,223 @@ import _ from 'underscore';
 import './css/foundation.css';
 import './css/style.css';
 
-console.log('it loaded!');
+import Trip from './app/models/trip';
+import TripList from './app/collections/trip_list';
 
-$(document).ready( () => {
-  $('main').html('<h1>Hello World!</h1>');
+const tripList = new TripList();
+tripList.fetch();
+
+let tripTemplate;
+const renderTrips = function renderTrips(tripList) {
+  $('.reservation-form').hide();
+  $('#trip-info').hide();
+  $('#trip-list').empty();
+  $('#filter-form').show();
+
+  tripList.forEach((trip) => {
+    $('#trip-list').append(tripTemplate(trip.attributes));
+  });
+  $('#trips-table').show();
+};
+
+let singleTripTemplate;
+const renderOneTrip = function renderOneTrip(id) {
+  $('#trip-info').empty();
+  $('#trip-info').show();
+  $('#new-trip-form').hide();
+  $('.reservation-form').show();
+
+  let trip;
+  trip = new Trip({id: id});
+  trip.fetch().done(() => {
+    $('#trip-info').append(singleTripTemplate(trip.attributes));
+  });
+};
+
+const tripFields = ['name', 'cost', 'weeks', 'continent', 'about', 'category'];
+
+const events = {
+  addTrip(event) {
+    event.preventDefault();
+    const tripData = {};
+    tripFields.forEach( (field) => {
+      let val;
+      if (field === 'continent') {
+        let val = $(`select[name=${field}]`).val();
+      } else {
+        let val = $(`input[name=${field}]`).val();
+      }
+      if (field === 'cost') {
+        val = parseFloat(val);
+      }
+      if (field === 'weeks') {
+        val = parseInt(val);
+      }
+      if (val != '') {
+        tripData[field] = val;
+      }
+    });
+
+    const trip = new Trip(tripData);
+    if (trip.isValid()) {
+      trip.save({}, {
+        success: events.successfulSave,
+        error: events.failedSave
+      });
+    }  else {
+      events.failedSave(trip);
```

Comment:
This should have `events.failedSave(trip, trip.validationError);` to output the validation errors.